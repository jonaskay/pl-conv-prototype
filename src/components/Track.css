.Track {
  height: 100%;
  overflow-x: hidden;
  overflow-y: auto;
  position: relative;
  width: 100%;
}

.Track-backdrop {
  background-position: center;
  background-size: cover;
  filter: blur(15px) brightness(.5);
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 0;
}

.Track-content {
  padding: 1rem;
  position: relative;
  text-align: center;
  z-index: 100;
}

.Track-img {
  box-shadow: 
    0 1px 5px 0 rgba(0,0,0,.2), 0 2px 2px 0 rgba(0,0,0,.14), 0 3px 1px -2px rgba(0,0,0,.12);
  display: block;
  height: 10rem; 
  margin: 1rem auto;
  width: 10rem
}
