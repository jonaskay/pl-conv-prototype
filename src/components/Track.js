import React from 'react';
import './Track.css';
import TrackItem from './TrackItem';

class Track extends React.Component {  
  render() {
    const track = this.props.track;

    return (
      <div className="Track">
        <div 
          className="Track-backdrop" 
          style={{backgroundImage: 'url(' + track.image + ')'}}></div>
        <div className="Track-content">
          <div style={{marginBottom: '1.25rem'}}>
            <img
              src={track.image} 
              alt={[track.name, track.artist].join(' - ')} 
              className="Track-img" />
            <div style={{fontSize: '1.125rem', marginBottom: '.25rem'}}>{track.name}</div>
            <div>{track.artist}</div>
          </div>
          <TrackItem service="spotify" resource_id={track.spotify_id} />
          <TrackItem service="youtube" resource_id={track.youtube_id} />
        </div>
      </div>
    );
  }
}

export default Track;
