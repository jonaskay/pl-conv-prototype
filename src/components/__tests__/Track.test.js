import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Track from '../Track';

it('renders a track', () => {
  const track = { 
    "id": 1,
    "name": "Bodak Yellow", 
    "artist": "Cardi B",
    "image": "https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc",
    "spotify_id": "2771LMNxwf62FTAdpJMQfM",
    "youtube_id": "PEGccV-NOm8"
  }
  const component = ReactTestRenderer.create(<Track track={track} />);
  expect(component).toMatchSnapshot();
});
