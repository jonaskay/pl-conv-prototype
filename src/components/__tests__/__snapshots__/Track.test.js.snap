// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a track 1`] = `
<div
  className="Track"
>
  <div
    className="Track-backdrop"
    style={
      Object {
        "backgroundImage": "url(https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc)",
      }
    }
  />
  <div
    className="Track-content"
  >
    <div
      style={
        Object {
          "marginBottom": "1.25rem",
        }
      }
    >
      <img
        alt="Bodak Yellow - Cardi B"
        className="Track-img"
        src="https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc"
      />
      <div
        style={
          Object {
            "fontSize": "1.125rem",
            "marginBottom": ".25rem",
          }
        }
      >
        Bodak Yellow
      </div>
      <div>
        Cardi B
      </div>
    </div>
    <div
      style={
        Object {
          "marginBottom": "1rem",
        }
      }
    >
      <iframe
        frameborder="0"
        height="80"
        src="https://open.spotify.com/embed?uri=spotify:track:2771LMNxwf62FTAdpJMQfM"
        title="Spotify Play Button"
        width="288"
      />
    </div>
    <div
      style={
        Object {
          "marginBottom": "1rem",
        }
      }
    >
      <iframe
        allow="autoplay; encrypted-media"
        allowFullScreen={true}
        frameborder="0"
        height="162"
        src="https://www.youtube.com/embed/PEGccV-NOm8?controls=0&amp;showinfo=0"
        title="YouTube Player"
        width="288"
      />
    </div>
  </div>
</div>
`;
